rllib/policy/policy.py

```diff
@@ -71,28 +73,44 @@
 #       "pol1": PolicySpec(None, Box, Discrete(2), {"lr": 0.0001}),
 #       "pol2": PolicySpec(config={"lr": 0.001}),
 #     }
-PolicySpec = PublicAPI(
-    namedtuple(
-        "PolicySpec",
-        [
-            # If None, use the Algorithm's default policy class stored under
-            # `Algorithm._policy_class`.
-            "policy_class",
-            # If None, use the env's observation space. If None and there is no Env
-            # (e.g. offline RL), an error is thrown.
-            "observation_space",
-            # If None, use the env's action space. If None and there is no Env
-            # (e.g. offline RL), an error is thrown.
-            "action_space",
-            # Overrides defined keys in the main Algorithm config.
-            # If None, use {}.
-            "config",
-        ],
-    )
-)  # defaults=(None, None, None, None)
-# TODO: From 3.7 on, we could pass `defaults` into the above constructor.
-#  We still support py3.6.
-PolicySpec.__new__.__defaults__ = (None, None, None, None)
+@PublicAPI
+class PolicySpec:
+    def __init__(
+        self, policy_class=None, observation_space=None, action_space=None, config=None
+    ):
+        # If None, use the Algorithm's default policy class stored under
+        # `Algorithm._policy_class`.
+        self.policy_class = policy_class
+        # If None, use the env's observation space. If None and there is no Env
+        # (e.g. offline RL), an error is thrown.
+        self.observation_space = observation_space
+        # If None, use the env's action space. If None and there is no Env
+        # (e.g. offline RL), an error is thrown.
+        self.action_space = action_space
+        # Overrides defined keys in the main Algorithm config.
+        # If None, use {}.
+        self.config = config
+
+    def serialize(self) -> Dict:
+        return {
+            # TODO(jungong) : try making the policy_class config durable.
```

Comment:
**suggestion:** Can you just save `f"{policy_class.__module__}.{policy_class.__name__}"`?

Reply:
yeah, that's the plan, although I am not 100% sure what happens if it's a user defined policy, or a policy defined in the launch script, etc.
so I thought I will punt this and come back after I demo connectors next Mon, so I have time to do better testing.